schemes/musica/micm/musica_ccpp_micm_util.F90

```diff
@@ -7,94 +7,69 @@ module micm_util
+    ! state.variables_[i_cell][i_species] = concentrations[i_species_elem++]
     i_elem = 1
     i_constituents = 1
-    i_rate_params = 1
     do i_layer = 1, num_layers
       do i_column = 1, num_columns
-        m_temperature(i_elem) = real(temperature(i_column, i_layer), c_double)
-        m_pressure(i_elem) = real(pressure(i_column, i_layer), c_double)
-        m_dry_air_density(i_elem) = real(dry_air_density(i_column, i_layer), c_double)
-        m_constituents(i_constituents : i_constituents + num_constituents - 1) &
+        micm_temperature(i_elem) = real(temperature(i_column, i_layer), c_double)
+        micm_pressure(i_elem) = real(pressure(i_column, i_layer), c_double)
+        micm_dry_air_density(i_elem) = real(dry_air_density(i_column, i_layer), c_double)
+        micm_constituents(i_constituents : i_constituents + num_constituents - 1) &
                             = real(constituents(i_column, i_layer, :), c_double)
-        m_rate_params(i_rate_params : i_rate_params + num_rate_params - 1) &
-                            = real(rate_params(i_column, i_layer, :), c_double)
         i_elem = i_elem + 1
         i_constituents = i_constituents + num_constituents
-        i_rate_params = i_rate_params + num_rate_params
       end do
     end do
 
   end subroutine reshape_into_micm_arr
 
-  subroutine reshape_into_ccpp_arr(temperature, pressure, dry_air_density, constituents, &
-      rate_params, m_temperature, m_pressure, m_dry_air_density, m_constituents, m_rate_params)
+  !> Reshape array (1D -> 3D) and convert type (c_double -> kind_phys)
+  subroutine reshape_into_ccpp_arr(micm_constituents, constituents)
     use iso_c_binding, only: c_double
     use ccpp_kinds,    only: kind_phys
-    real(kind_phys), intent(out) :: temperature(:,:)     ! K
-    real(kind_phys), intent(out) :: pressure(:,:)        ! Pa
-    real(kind_phys), intent(out) :: dry_air_density(:,:) ! kg m-3
-    real(kind_phys), intent(out) :: constituents(:,:,:)  ! kg kg-1
-    real(kind_phys), intent(out) :: rate_params(:,:,:)
-    real(c_double),  intent(in)  :: m_temperature(:)     ! K
-    real(c_double),  intent(in)  :: m_pressure(:)        ! Pa
-    real(c_double),  intent(in)  :: m_dry_air_density(:) ! kg m-3
-    real(c_double),  intent(in)  :: m_constituents(:)    ! kg kg-1
-    real(c_double),  intent(in)  :: m_rate_params(:)
+
+    real(c_double),  intent(in)    :: micm_constituents(:) ! kg kg-1
+    real(kind_phys), intent(inout) :: constituents(:,:,:)  ! kg kg-1
 
     ! local variables
-    integer :: num_columns, num_layers
-    integer :: num_constituents, num_rate_params
-    integer :: i_column, i_layer, i_elem, i_constituents, i_rate_params
+    integer :: num_columns, num_layers, num_constituents
+    integer :: i_column, i_layer, i_constituents
 
     num_columns = size(constituents, dim=1)
     num_layers = size(constituents, dim=2)
     num_constituents = size(constituents, dim=3)
```

Comment:
```suggestion
    num_columns = size(micm_constituents, dim=1)
    num_layers = size(micm_constituents, dim=2)
    num_constituents = size(micm_constituents, dim=3)
```

Should it be this instead?

Reply:
Yeah I found this possibly confusing. The function `reshape_into_ccpp_arr()` takes 1-D micm array and reshape it into 3-D CCPP array. It doesn't seem straightforward to get the dimension of CCPP array from micm array because this function does not want to know the number of columns and layers that are set when MCIM species are registered. (unless we want to do this way) I made CCPP array with `intent(inout)` instead of `intent(out)` solely to get the dimensions for the array. Any thoughts on this? I was also questioning about this. 
```Fortran
!> Reshape array (1D -> 3D) and convert type (c_double -> kind_phys)
subroutine reshape_into_ccpp_arr(micm_constituents, constituents)
```

Reply:
Ah, I understand now. I don't know of a better way